common/root-tree-utils.c

```diff
@@ -212,3 +214,266 @@ int btrfs_link_subvolume(struct btrfs_trans_handle *trans,
 	btrfs_abort_transaction(trans, ret);
 	return ret;
 }
+
+static int empty_tree(struct btrfs_root *root)
+{
+	struct btrfs_trans_handle *trans;
+	struct btrfs_path path = { 0 };
+	struct btrfs_key key = { 0 };
+	int ret;
+
+	trans = btrfs_start_transaction(root, 1);
+	if (IS_ERR(trans)) {
+		ret = PTR_ERR(trans);
+		errno = -ret;
+		error_msg(ERROR_MSG_START_TRANS, "empty tree: %m");
+		return ret;
+	}
+	while (true) {
+		int nr_items;
+
+		ret = btrfs_search_slot(trans, root, &key, &path, -1, 1);
+		if (ret < 0) {
+			errno = -ret;
+			error("failed to locate the first key of root %lld: %m",
```

Comment:
You can use longer lines in code that is not synced with kernel (ie anything besides kernel-shared/), up to 90 is ok, namely for the error messages as they're not critical code.

Reply:
Well, it's just a coincident, I take not special care to shorten the strings at all, didn't even notice it ends exactly at 80 chars.
And I have no extra into to add to that error message either.